src/Adapter/OrderReturn/CommandHandler/UpdateOrderReturnStateHandler.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\CommandHandler;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\AbstractOrderReturnHandler;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\CommandHandler\UpdateOrderReturnStateHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnOrderStateConstraintException;
+
+class UpdateOrderReturnStateHandler extends AbstractOrderReturnHandler implements UpdateOrderReturnStateHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateOrderReturnStateCommand $command): void
+    {
+        $orderReturnId = $command->getOrderReturnId();
+
+        $orderReturn = $this->getOrderReturn($orderReturnId);
+
+        $orderReturn = $this->updateOrderReturnWithCommandData($orderReturn, $command);
+
+        $this->assertRequiredFieldsAreNotMissing($orderReturn);
+
+        if (false === $orderReturn->validateFields(false)) {
+            throw new OrderReturnException('Order return contains invalid field values');
+        }
+
+        if (false === $orderReturn->update()) {
```

Comment:
Object model write operations like add/update/save should be wrapped in try-catch to disallow old PrestaShopException bubbling up. Once again this is also covered in AbstractObjectModelRepository that you could use